src/semantic-router/pkg/extproc/extproc_test.go

```diff
@@ -2922,26 +2922,32 @@ func TestSetReasoningModeToRequestBody(t *testing.T) {
 					t.Fatalf("Expected non-empty chat_template_kwargs")
 				}
 
-				// Validate the specific parameter based on model type
-				switch tc.model {
-				case "deepseek-v31", "ds-1.5b":
-					if thinkingValue, exists := kwargs["thinking"]; !exists {
-						t.Fatalf("Expected 'thinking' parameter in chat_template_kwargs for DeepSeek model")
-					} else if thinkingValue != true {
-						t.Fatalf("Expected 'thinking' to be true, got %v", thinkingValue)
+				// Validate the specific parameter for chat_template_kwargs families.
+				// (Different families use different parameter names, but the value should match tc.enabled.)
+				if v, exists := kwargs["thinking"]; exists {
+					if v != tc.enabled {
+						t.Fatalf("Expected chat_template_kwargs.thinking to be %v, got %v", tc.enabled, v)
 					}
-				case "qwen3-7b":
-					if thinkingValue, exists := kwargs["enable_thinking"]; !exists {
-						t.Fatalf("Expected 'enable_thinking' parameter in chat_template_kwargs for Qwen3 model")
-					} else if thinkingValue != true {
-						t.Fatalf("Expected 'enable_thinking' to be true, got %v", thinkingValue)
+				} else if v, exists := kwargs["enable_thinking"]; exists {
+					if v != tc.enabled {
+						t.Fatalf("Expected chat_template_kwargs.enable_thinking to be %v, got %v", tc.enabled, v)
 					}
+				} else {
+					t.Fatalf("Expected chat_template_kwargs to contain either 'thinking' or 'enable_thinking', got keys=%v", mapKeys(kwargs))
 				}
 			}
 		})
 	}
 }
 
+func mapKeys(m map[string]interface{}) []string {
```

Comment:
Nit: mapKeys suggests using sort.Strings(keys) to make failure messages more stable and readable.